ts/src/header-validator/validate-json.ts

```diff
@@ -1208,8 +1212,34 @@ function aggregatableKeyValue(
     )
 }
 
-function aggregatableValues(ctx: Context, j: Json): Maybe<Map<string, number>> {
-  return keyValues(ctx, j, aggregatableKeyValue)
+function aggregatableValues(ctx: Context, j: Json): Maybe<AggregatableValue[]> {
+  return typeSwitch(ctx, j, {
+    object: (ctx, j) =>
+      struct(
```

Comment:
I don't think it's correct to use `struct` here: If `j` is an object, then `j` directly contains the key-value pairs. I'd keep `function aggregatableValues` but rename it to `function aggregatableKeyValues` and then make this function like:

```typescript
function aggregatablevalues(ctx: Context, j: Json): Maybe<AggregatableValue[]> {
  return typeSwitch(ctx, j, {
    object: (ctx, j) => aggregatableKeyValues(ctx, j).map((values) => ({ values, filters: [], notFilters: []})),
    list: ... // what you currently have, but using `aggregatableKeyValues` instead of `keyValues`
```